motionplan/motionPlanner.go

```diff
@@ -54,10 +54,57 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
+	if req.Logger == nil {
+		req.Logger = golog.NewLogger("plan-request-logger")
+	}
+
+	if req.Frame == nil {
+		return errors.New("PlanRequest cannot have nil frame")
+	}
+
+	if req.FrameSystem == nil {
+		req.Logger.Debug("no FrameSystem was given, so we ephemerally create a frame system containing just the frame for the solve")
+		fs := frame.NewEmptyFrameSystem("")
```

Comment:
I don't think a function with `validate` in the name should be mutating the request. Validation should be a read-only operation which.

Why do we need to do this?

If we need to set defaults on the request I think we should put that logic in a different function with more appropriate name. 

We should decouple default setting (a mutation) from validation (read only).

Reply:
> I don't think a function with `validate` in the name should be mutating the request. Validation should be a read-only operation which.

I agree that `validate` is a misnomer if we are mutating the object the receive.
I will remove mutations entirely.

> Why do we need to do this?

We need to know how to move the frame around in the world so that it reaches the goal. We know the frame's relation to the world it occupies through the framesystem.